core/src/main/java/io/ddf/DDF.java

```diff
@@ -346,6 +346,12 @@ private void validateSchema(Schema schema) throws DDFException {
         if (columnSet.contains(column.getName())) {
           throw new DDFException(String.format("Duplicated column name %s", column.getName()));
         } else {
+          Pattern p = Pattern.compile("^[a-zA-Z0-9_-]*$");
```

Comment:
Should we force the user to start with character and end with character / digit. E.g: these colnames are considered invalid:
_colname
-colname
colname_
colname-

BA doesn't allow such colnames (but one conflict thing is some auto-generated colname is _c0, _c1....)

@nhanitvn @PangZhi ?